src/Twig/SourceCodeExtension.php

```diff
@@ -117,20 +118,20 @@ private function getTemplateSource($template): array
      */
     private function unindentCode(string $code): string
     {
-        $formattedCode = $code;
-        $codeLines = explode("\n", $code);
+        $codeLines = u($code)->split("\n");
 
-        $indentedLines = array_filter($codeLines, function ($lineOfCode) {
-            return '' === $lineOfCode || 0 === mb_strpos($lineOfCode, '    ');
+        $indentedOrBlankLines = array_filter($codeLines, function ($lineOfCode) {
+            return u($lineOfCode)->isEmpty() || u($lineOfCode)->startsWith('    ');
```

Comment:
you should probably instantiate only one string object and reuse it.

Reply:
For an application where performance is critical, I'd agree ... but this is only used when checking the source code explicitly, so it rarely happens. Let's keep the code a bit simpler. Thanks.